common/com/twitter/intellij/pants/util/PantsUtil.java

```diff
@@ -950,4 +976,10 @@ public static void invalidatePluginCaches() {
   public static List<String> parseCmdParameters(Optional<String> cmdArgsLine) {
     return cmdArgsLine.map(ParametersListUtil::parse).orElse(ContainerUtil.newArrayList());
   }
+
+  public static <T, U> Set<U> lookupKeys(Set<T> keys, ImmutableMap<T, ImmutableSet<U>> map) {
```

Comment:
Could you comment and test for this? It is unclear to me what it tries to do.

Reply:
Yes. This bit (along with the changes in `PantsConstants` and `PantsMakeBeforeRun::executeTask`) was written so that the right options get added to the command line for the requested set of tasks. For example, adding `--async` to the command line causes pants to error out unless `clean-all` is one of the tasks. This wasn't an issue before when we were only compiling because we could explicitly check whether we were also running `clean-all`. I'll add comments and tests for this so it's more clear.